lib/reline/io/windows.rb

```diff
@@ -350,9 +351,12 @@ def get_console_screen_buffer_info
 
   def get_screen_size
     unless csbi = get_console_screen_buffer_info
-      return [1, 1]
+      return [24, 80]
     end
-    csbi[0, 4].unpack('SS').reverse
+    top = csbi[12, 2].unpack1('S')
+    bottom = csbi[16, 2].unpack1('S')
+    width = csbi[0, 2].unpack1('S')
+    [bottom - top + 1, width]
```

Comment:
My understanding by reading the [document](https://learn.microsoft.com/windows/console/console-screen-buffer-info-str) is:
```
hidden_screen_buffer01
hidden_screen_buffer02
hidden_screen_buffer03
hidden_screen_buffer04
visible_screen_buffer1
visible_screen_buffer2
visible_sc█een_buffer3
visible_screen_buffer4

screen_buffer width = 22, screen_buffer height = 8
cursor x = 10 y = 6 (cursor coord in screen buffer, not coord in visible window)
window left=0 right=21 top=4 bottom=7
```
Is this right? If so, I think `def cursor_pos` should use `CursorPos.new(x, y - top)`. Can you check this?

Reply:
You are right.
I tried to modify cursor_pos, but even the cursor moving functions needed to take csbi.srWindow.Top into consideration. Needs some more work.

Reply:
done in df7054193d53d0b55f7787d2420af59c284c64c8 .

Reply:
Thank you :+1: